### ISSUE TITLE
Compile Error: -Werror=old-style-cast

### ISSUE THREAD

**Issue by Mako88** [dxx-rebirth/dxx-rebirth#213]
Looks like you may have missed a couple Kp ;)
```
In file included from common/include/ogl_init.h:21:0,
                 from common/include/internal.h:12,
                 from similar/arch/ogl/gr.cpp:54:
common/include/loadgl.h: In function ‘void* dll_LoadModule(const char*)’:
common/include/loadgl.h:283:17: error: use of old-style cast [-Werror=old-style-cast]
  return (void *)handle;
                 ^
common/include/loadgl.h: In function ‘void dll_UnloadModule(void*)’:
common/include/loadgl.h:288:22: error: use of old-style cast [-Werror=old-style-cast]
  handle = (HINSTANCE)hdl;
                      ^
common/include/loadgl.h: In function ‘void* dll_GetSymbol(void*, const char*)’:
common/include/loadgl.h:299:43: error: use of old-style cast [-Werror=old-style-cast]
  return (void *)GetProcAddress((HINSTANCE)dllhandle,symname);
                                           ^
common/include/loadgl.h:299:60: error: use of old-style cast [-Werror=old-style-cast]
  return (void *)GetProcAddress((HINSTANCE)dllhandle,symname);
                                                            ^
common/include/loadgl.h: In function ‘bool OpenGL_LoadLibrary(bool, const char*)’:
common/include/loadgl.h:370:80: error: use of old-style cast [-Werror=old-style-cast]
 dglAlphaFunc = (glAlphaFunc_fp)dll_GetSymbol(OpenGLModuleHandle,"glAlphaFunc");
                                                                              ^
common/include/loadgl.h:371:86: error: use of old-style cast [-Werror=old-style-cast]
 dTexture = (glBindTexture_fp)dll_GetSymbol(OpenGLModuleHandle,"glBindTexture");
                                                                              ^
common/include/loadgl.h:372:80: error: use of old-style cast [-Werror=old-style-cast]
 dglBlendFunc = (glBlendFunc_fp)dll_GetSymbol(OpenGLModuleHandle,"glBlendFunc");
                                                                              ^
common/include/loadgl.h:373:68: error: use of old-style cast [-Werror=old-style-cast]
   dglClear = (glClear_fp)dll_GetSymbol(OpenGLModuleHandle,"glClear");
                                                                    ^
common/include/loadgl.h:374:83: error: use of old-style cast [-Werror=old-style-cast]
 ClearColor = (glClearColor_fp)dll_GetSymbol(OpenGLModuleHandle,"glClearColor");
                                                                              ^
common/include/loadgl.h:375:74: error: use of old-style cast [-Werror=old-style-cast]
   dglColor4f = (glColor4f_fp)dll_GetSymbol(OpenGLModuleHandle,"glColor4f");
                                                                          ^
common/include/loadgl.h:376:89: error: use of old-style cast [-Werror=old-style-cast]
 ointer = (glColorPointer_fp)dll_GetSymbol(OpenGLModuleHandle,"glColorPointer");
                                                                              ^
common/include/loadgl.h:377:77: error: use of old-style cast [-Werror=old-style-cast]
   dglCullFace = (glCullFace_fp)dll_GetSymbol(OpenGLModuleHandle,"glCullFace");
                                                                             ^
common/include/loadgl.h:378:95: error: use of old-style cast [-Werror=old-style-cast]
 es = (glDeleteTextures_fp)dll_GetSymbol(OpenGLModuleHandle,"glDeleteTextures");
                                                                              ^
common/include/loadgl.h:379:80: error: use of old-style cast [-Werror=old-style-cast]
 dglDepthFunc = (glDepthFunc_fp)dll_GetSymbol(OpenGLModuleHandle,"glDepthFunc");
                                                                              ^
common/include/loadgl.h:380:80: error: use of old-style cast [-Werror=old-style-cast]
 dglDepthMask = (glDepthMask_fp)dll_GetSymbol(OpenGLModuleHandle,"glDepthMask");
                                                                              ^
common/include/loadgl.h:381:74: error: use of old-style cast [-Werror=old-style-cast]
   dglDisable = (glDisable_fp)dll_GetSymbol(OpenGLModuleHandle,"glDisable");
                                                                          ^
common/include/loadgl.h:382:107: error: use of old-style cast [-Werror=old-style-cast]
 DisableClientState_fp)dll_GetSymbol(OpenGLModuleHandle,"glDisableClientState");
                                                                              ^
common/include/loadgl.h:383:83: error: use of old-style cast [-Werror=old-style-cast]
 DrawArrays = (glDrawArrays_fp)dll_GetSymbol(OpenGLModuleHandle,"glDrawArrays");
                                                                              ^
common/include/loadgl.h:384:71: error: use of old-style cast [-Werror=old-style-cast]
   dglEnable = (glEnable_fp)dll_GetSymbol(OpenGLModuleHandle,"glEnable");
                                                                       ^
common/include/loadgl.h:385:104: error: use of old-style cast [-Werror=old-style-cast]
 glEnableClientState_fp)dll_GetSymbol(OpenGLModuleHandle,"glEnableClientState");
                                                                              ^
common/include/loadgl.h:386:62: error: use of old-style cast [-Werror=old-style-cast]
   dglEnd = (glEnd_fp)dll_GetSymbol(OpenGLModuleHandle,"glEnd");
                                                              ^
common/include/loadgl.h:387:71: error: use of old-style cast [-Werror=old-style-cast]
   dglFinish = (glFinish_fp)dll_GetSymbol(OpenGLModuleHandle,"glFinish");
                                                                       ^
common/include/loadgl.h:388:86: error: use of old-style cast [-Werror=old-style-cast]
 Textures = (glGenTextures_fp)dll_GetSymbol(OpenGLModuleHandle,"glGenTextures");
                                                                              ^
common/include/loadgl.h:389:80: error: use of old-style cast [-Werror=old-style-cast]
 dglGetFloatv = (glGetFloatv_fp)dll_GetSymbol(OpenGLModuleHandle,"glGetFloatv");
                                                                              ^
common/include/loadgl.h:390:86: error: use of old-style cast [-Werror=old-style-cast]
 Integerv = (glGetIntegerv_fp)dll_GetSymbol(OpenGLModuleHandle,"glGetIntegerv");
                                                                              ^
common/include/loadgl.h:391:80: error: use of old-style cast [-Werror=old-style-cast]
 dglGetString = (glGetString_fp)dll_GetSymbol(OpenGLModuleHandle,"glGetString");
                                                                              ^
common/include/loadgl.h:392:119: error: use of old-style cast [-Werror=old-style-cast]
 velParameteriv_fp)dll_GetSymbol(OpenGLModuleHandle,"glGetTexLevelParameteriv");
                                                                              ^
common/include/loadgl.h:393:65: error: use of old-style cast [-Werror=old-style-cast]
   dglHint = (glHint_fp)dll_GetSymbol(OpenGLModuleHandle,"glHint");
                                                                 ^
common/include/loadgl.h:394:80: error: use of old-style cast [-Werror=old-style-cast]
 dglLineWidth = (glLineWidth_fp)dll_GetSymbol(OpenGLModuleHandle,"glLineWidth");
                                                                              ^
common/include/loadgl.h:395:89: error: use of old-style cast [-Werror=old-style-cast]
 entity = (glLoadIdentity_fp)dll_GetSymbol(OpenGLModuleHandle,"glLoadIdentity");
                                                                              ^
common/include/loadgl.h:396:83: error: use of old-style cast [-Werror=old-style-cast]
 MatrixMode = (glMatrixMode_fp)dll_GetSymbol(OpenGLModuleHandle,"glMatrixMode");
                                                                              ^
common/include/loadgl.h:397:68: error: use of old-style cast [-Werror=old-style-cast]
   dglOrtho = (glOrtho_fp)dll_GetSymbol(OpenGLModuleHandle,"glOrtho");
                                                                    ^
common/include/loadgl.h:398:80: error: use of old-style cast [-Werror=old-style-cast]
 dglPointSize = (glPointSize_fp)dll_GetSymbol(OpenGLModuleHandle,"glPointSize");
                                                                              ^
common/include/loadgl.h:399:80: error: use of old-style cast [-Werror=old-style-cast]
 dglPopMatrix = (glPopMatrix_fp)dll_GetSymbol(OpenGLModuleHandle,"glPopMatrix");
                                                                              ^
common/include/loadgl.h:400:107: error: use of old-style cast [-Werror=old-style-cast]
 PrioritizeTextures_fp)dll_GetSymbol(OpenGLModuleHandle,"glPrioritizeTextures");
                                                                              ^
common/include/loadgl.h:401:83: error: use of old-style cast [-Werror=old-style-cast]
 PushMatrix = (glPushMatrix_fp)dll_GetSymbol(OpenGLModuleHandle,"glPushMatrix");
                                                                              ^
common/include/loadgl.h:402:83: error: use of old-style cast [-Werror=old-style-cast]
 ReadBuffer = (glReadBuffer_fp)dll_GetSymbol(OpenGLModuleHandle,"glReadBuffer");
                                                                              ^
common/include/loadgl.h:403:83: error: use of old-style cast [-Werror=old-style-cast]
 ReadPixels = (glReadPixels_fp)dll_GetSymbol(OpenGLModuleHandle,"glReadPixels");
                                                                              ^
common/include/loadgl.h:404:71: error: use of old-style cast [-Werror=old-style-cast]
   dglScalef = (glScalef_fp)dll_GetSymbol(OpenGLModuleHandle,"glScalef");
                                                                       ^
common/include/loadgl.h:405:83: error: use of old-style cast [-Werror=old-style-cast]
 ShadeModel = (glShadeModel_fp)dll_GetSymbol(OpenGLModuleHandle,"glShadeModel");
                                                                              ^
common/include/loadgl.h:406:98: error: use of old-style cast [-Werror=old-style-cast]
  = (glTexCoordPointer_fp)dll_GetSymbol(OpenGLModuleHandle,"glTexCoordPointer");
                                                                              ^
common/include/loadgl.h:407:74: error: use of old-style cast [-Werror=old-style-cast]
   dglTexEnvi = (glTexEnvi_fp)dll_GetSymbol(OpenGLModuleHandle,"glTexEnvi");
                                                                          ^
common/include/loadgl.h:408:83: error: use of old-style cast [-Werror=old-style-cast]
 TexImage2D = (glTexImage2D_fp)dll_GetSymbol(OpenGLModuleHandle,"glTexImage2D");
                                                                              ^
common/include/loadgl.h:409:92: error: use of old-style cast [-Werror=old-style-cast]
 terf = (glTexParameterf_fp)dll_GetSymbol(OpenGLModuleHandle,"glTexParameterf");
                                                                              ^
common/include/loadgl.h:410:92: error: use of old-style cast [-Werror=old-style-cast]
 teri = (glTexParameteri_fp)dll_GetSymbol(OpenGLModuleHandle,"glTexParameteri");
                                                                              ^
common/include/loadgl.h:411:83: error: use of old-style cast [-Werror=old-style-cast]
 Translatef = (glTranslatef_fp)dll_GetSymbol(OpenGLModuleHandle,"glTranslatef");
                                                                              ^
common/include/loadgl.h:412:92: error: use of old-style cast [-Werror=old-style-cast]
 nter = (glVertexPointer_fp)dll_GetSymbol(OpenGLModuleHandle,"glVertexPointer");
                                                                              ^
common/include/loadgl.h:413:77: error: use of old-style cast [-Werror=old-style-cast]
   dglViewport = (glViewport_fp)dll_GetSymbol(OpenGLModuleHandle,"glViewport");
                                                                             ^
```

**Comment by vLKp** [comment]
This is strange to me.  I know I ran at least one mingw32 build because that prompted me to remove more unused symbols from `common/include/loadgl.h`.  I know I tested with my standard set of configurations before pushing.  I see that a bug I fixed yesterday in the SConstruct handling could have caused it to skip building some targets, which might explain how I did not retest this configuration.  However, I thought that I had run a mingw32 only build.
